core/src/components/checkbox/sdds-checkbox.scss

```diff
@@ -127,5 +127,40 @@
         }
       }
     }
+
+    &:indeterminate {
+      &::after {
+        background-image: var(--sdds-checkbox-background-img-indeterminate);
+        background-color: var(--sdds-checkbox-interaction-01);
+        background-repeat: no-repeat;
+        background-position: center;
+      }
+
+      &:hover {
+        &::before {
+          background-color: var(--sdds-checkbox-background-hover);
+          opacity: var(--sdds-checkbox-background-opacity-hover);
+        }
+      }
+
+      &:disabled,
+      &.disabled {
+        &:hover::before,
+        &::before {
+          display: block;
+          width: 16px;
+          height: 16px;
+          left: 4px;
+          top: 4px;
+          border: 1px solid #b0b7c4;
```

Comment:
I think a lot of these styles should be for both checked/indeterminate state. Could reduce the amount of css